plugins/org.python.pydev.parser/src/org/python/pydev/parser/grammar310/python.jjt_template

```diff
@@ -0,0 +1,1502 @@
+{ $RAISE [test() [{grammarActions.addSpecialToken(" from ");} <FROM> test()]] }
+
+
+$IMPORT_STMT
+
+
+Import Import(): {}
+{ dotted_as_name() ($COMMA dotted_as_name())*
+  {return (Import)jjtree.peekNode();}
+}
+
+//import_from: ('from' ('.'* dotted_name | '.'+)
+//              'import' ('*' | '(' import_as_names ')' | import_as_names))
+void ImportFrom(): { int level=0; int state=0;String fromName=null;String importName=null; }
+{
+    //we need to set the  $IMPORT in both otherwise the lookahead will not work as we want it to work
+    //because it confuses the import with the dotted name
+     (("." {level++;} )* (fromName=dotted_name())? )  {if(fromName==null && level==0){throw new ParseException("Expecting to find '.' or name in import.");}}
+     $IMPORT
+
+    (
+        //from xxx import *
+        <MULTIPLY> {grammarActions.addSpecialToken("*",STRATEGY_ADD_AFTER_PREV);}//from xx import *
+
+        //from xxx import a,b,c
+        | (importName=import_as_name() {if(fromName != null && fromName.equals("__future__"))handleFutureImports(importName);}
+            ($COMMA (importName=import_as_name()){if(fromName != null && fromName.equals("__future__"))handleFutureImports(importName);})*
+          )
+
+        //from xxx import (a,b,c)
+        | $LPAREN1
+          (importName=import_as_name()){if(fromName != null && fromName.equals("__future__"))handleFutureImports(importName);}
+          (
+           ({
+             if(state!=0){
+                 throw new ParseException("Invalid syntax: 2 commas cannot be grouped.", getToken(1));
+             }
+             state=1;
+             }
+             $COMMA ( {state=0;} (importName=import_as_name(){if(fromName != null && fromName.equals("__future__"))handleFutureImports(importName);}))? )*
+           $RPAREN
+          )
+    )
+    //now, let's set the correct level for the module
+    {grammarActions.setImportFromLevel(level);}
+}
+
+//dotted_as_name: dotted_name [NAME NAME]
+void dotted_as_name(): {}
+{ dotted_name() [$AS Name()] }
+
+$DOTTED_NAME
+
+//import_as_name: NAME [NAME NAME]
+String import_as_name(): { Token t; }
+{ t=Name() [$AS Name()] { return t.image; } }
+
+//global_stmt: 'global' NAME (',' NAME)*
+void global_stmt(): {}
+{ $GLOBAL Name() ($COMMA Name())* }
+
+
+//nonlocal_stmt: 'nonlocal' NAME (',' NAME)* [','] ['=' testlist]
+void nonlocal_stmt(): {}
+{ <NONLOCAL> {grammarActions.addSpecialToken("nonlocal ", STRATEGY_BEFORE_NEXT);} Name() (LOOKAHEAD(2)$COMMA Name())* [$COMMA] [$EQUAL2 testlist()]}
+
+
+$ASSERT
+
+
+// star_named_expr: [* **] named_expr
+void star_named_expr() #void : {}
+{
+    (<MULTIPLY> | <POWER>) namedexpr_test()
+}
+
+// subject_expr: (star_named_expr (, star_named_expr)*) | named_expr
+void subject_expr() #void : {}
+{
+	(star_named_expr() (<COMMA> star_named_expr())*) | namedexpr_test()
+}
+
+// case_block: "case" named_expr_test : suite
+void case_block(): {}
+{
+	<CASE> {this.markLastAsSuiteStart();} namedexpr_test() <COLON> suite()
```

Comment:
We can't really reuse the `namedexpr_test()` here (this will be much more complex -- see the related changes for Python at: https://github.com/python/cpython/commit/145bf269df3530176f6ebeab1324890ef7070bf8#diff-2973ca53337859793077e9bdc1a1623063379f0fdfcb788836fd82ebb66b763bR211).